packages/docusaurus-theme-classic/src/theme/BlogListPage/StructuredData/index.tsx

```diff
@@ -0,0 +1,95 @@
+ */
+
+import React from 'react';
+import {useBaseUrlUtils} from '@docusaurus/useBaseUrl';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import type {Props} from '@theme/BlogListPage/StructuredData';
+
+export default function BlogListPageStructuredData(props: Props): JSX.Element {
+  const {siteConfig} = useDocusaurusContext();
+  const {withBaseUrl} = useBaseUrlUtils();
+
+  const {
+    metadata: {blogDescription, blogTitle, permalink},
+  } = props;
+
+  const url = `${siteConfig.url}${permalink}`;
+
+  // details on structured data support: https://schema.org/Blog
+  const blogStructuredData = {
+    '@context': 'https://schema.org',
+    '@type': 'Blog',
+    '@id': url,
+    mainEntityOfPage: url,
+    headline: blogTitle,
+    description: blogDescription,
+    blogPost: props.items.map((blogItem) => {
+      const {
+        content: {assets, frontMatter, metadata},
+      } = blogItem;
+      const {date, title, description} = metadata;
+
+      const image = assets.image ?? frontMatter.image;
+      const keywords = frontMatter.keywords ?? [];
+
+      // an array of https://schema.org/Person
+      const authorsStructuredData = metadata.authors.map((author) => ({
+        '@type': 'Person',
+        ...(author.name ? {name: author.name} : {}),
+        ...(author.title ? {description: author.title} : {}),
+        ...(author.url ? {url: author.url} : {}),
+        ...(author.email ? {email: author.email} : {}),
+        ...(author.imageURL ? {image: author.imageURL} : {}),
+      }));
+
+      const blogUrl = `${siteConfig.url}${metadata.permalink}`;
+      const imageUrl = image ? withBaseUrl(image, {absolute: true}) : undefined;
+
+      return {
+        '@type': 'BlogPosting',
+        '@id': blogUrl,
+        mainEntityOfPage: blogUrl,
+        url: blogUrl,
+        headline: title,
+        name: title,
+        description,
+        datePublished: date,
+        author:
+          authorsStructuredData.length === 1
+            ? authorsStructuredData[0]
+            : authorsStructuredData,
+        ...(image
+          ? {
+              // details on structured data support: https://schema.org/ImageObject
+              image: {
+                '@type': 'ImageObject',
+                '@id': imageUrl,
+                url: imageUrl,
+                contentUrl: imageUrl,
+                caption: `title image for the blog post: ${title}`,
+              },
+            }
+          : {}),
+        ...(keywords ? {keywords} : {}),
+      };
+    }),
+  };
+
+  return (
+    <script
+      type="application/ld+json"
+      // We're using dangerouslySetInnerHTML because we want to avoid React
+      // transforming quotes into &quot; which upsets parsers.
+      // The entire contents is a stringified JSON object so it is safe
+      // eslint-disable-next-line react/no-danger
+      dangerouslySetInnerHTML={{
+        __html: JSON.stringify(blogStructuredData),
+      }}
+    />
+  );
+}
```

Comment:
Maybe we'd want to create a helper component in theme-common to encapsulate this logic? 

We might use structured data in other places in the future, and we probably don't want to copy/paste this solution everywhere

Reply:
Makes sense - will do

Reply:
Created a dedicated `StructuredData` component